app/mailers/application_mailer.rb

```diff
@@ -64,48 +47,114 @@ def can_group_notify(data, recipient)
       (can_notify(recipient, 'meeting_notify') && can_meeting(data))
   end
 
-  # rubocop:disable MethodLength
   def group_notify(data, recipient)
-    @data = data
     @recipient = recipient
-
-    if @data['type'] == 'new_group'
-      subject = new_group_subject(@data)
-      @message = new_group_body(@data)
-    elsif @data['type'] == 'new_group_member'
-      subject = new_group_member_subject(@data)
-      @message = add_remove_group_leader_body(@data)
-    elsif @data['type'] == 'add_group_leader' && you?(@recipient, @data)
-      subject = add_group_leader_you_subject(@data)
-      @message = add_remove_group_leader_body(@data)
-    elsif @data['type'] == 'add_group_leader'
-      subject = add_group_leader_subject(@data)
-      @message = add_remove_group_leader_body(@data)
-    elsif @data['type'] == 'remove_group_leader' && you?(@recipient, @data)
-      subject = remove_group_leader_you_subject(@data)
-      @message = add_remove_group_leader_body(@data)
-    elsif @data['type'] == 'remove_group_leader'
-      subject = remove_group_leader_subject(@data)
-      @message = add_remove_group_leader_body(@data)
-    elsif @data['type'] == 'new_meeting'
-      subject = new_meeting_subject(@data)
-      @message = meeting_body(@data) + new_meeting_link(@data)
-    elsif @data['type'] == 'update_meeting'
-      subject = update_meeting_subject(@data)
-      @message = meeting_body(@data) + update_meeting_link(@data)
-    elsif @data['type'] == 'remove_meeting'
-      subject = remove_meeting_subject(@data)
-      @message = add_remove_group_leader_body(@data)
-    elsif @data['type'] == 'join_meeting'
-      subject = join_meeting_subject(@data)
-      @message = join_meeting_body(@data)
+    case data['type']
+    when 'new_group', 'new_group_member'
+      new_group_notify(data)
+    when 'add_group_leader'
+      add_group_notify(data, recipient)
+    when 'remove_group_leader' 
+      remove_group_notify(data, recipient)
+    when 'new_meeting'
+      new_meeting_group_notify(data)
+    when 'update_meeting'
+      update_meeting_group_notify(data)
+    when 'remove_meeting'
+      remove_meeting_group_notify(data)
+    when 'join_meeting'
+      join_meeting_group_notify(data)
+    else
+      @message = nil
```

Comment:
Do we need this else case? Will `@message` be implicitly nil?

Reply:
@message will definitely be implicitly nil. The main reason this is included is for the sake of being thorough and readability.

Also, IntelliJ showed a warning that there was a case statement without an else block. 

I'm happy to remove it, if you prefer a more succinct style

Reply:
Keeping it is fine then! Thanks for explaining!